sirepo/pkcli/job_agent.py

```diff
@@ -633,3 +670,8 @@ def __init__(self, stream):
         assert l < self._MAX, \
             'len(bytes)={} greater than _MAX={}'.format(l, _MAX)
         self.text.extend(t)
+
+
+def _terminate(dispatcher):
+    pkio.unchecked_remove(_PID_FILE)
```

Comment:
change the order of these two lines. If terminate gets in a loop (for whatever reason), then we'll want to pid to be killed.